### PR TITLE
[tripy] Fix gather operation when axis > 0

### PR DESCRIPTION
**Pull request by parthchadha**
Note: https://github.com/NVIDIA/TensorRT-Incubator/issues/52 will add more integration tests for Gather operation.

Fixes two issues with Gather op:
1. `FlatIRTensor` constructor was not provided `rank`.
2. When `axis>0`, slicing second half of shape tensor was incorrect.

### CODE REVIEW

**Comment by parthchadha** [comment]
```
======================================= 1968 passed, 73 skipped, 1 deselected, 1 warning in 620.30s (0:10:20) =======================================
```

File: tripy/tripy/frontend/trace/ops/gather.py [hunk 1]
```diff
@@ -110,10 +111,10 @@ def gather(input: "tripy.Tensor", dim: int, index: "tripy.Tensor") -> "tripy.Ten
...
         :caption: Example
 
-        data = tp.iota((3, 2, 2))
+        data = tp.iota((3, 3, 2))
         indices = tp.Tensor([0, 2], dtype=tp.int32)
-        output = tp.gather(data, 0, indices)
+        output = tp.gather(data, 1, indices)
 
-        assert np.array_equal(cp.from_dlpack(output).get(), np.take(cp.from_dlpack(data).get(), cp.from_dlpack(indices).get(), axis=0))
+        assert np.array_equal(cp.from_dlpack(output).get(), np.take(cp.from_dlpack(data).get(), cp.from_dlpack(indices).get(), axis=1))
```
Comment (pranavm-nvidia) [inline tripy/tripy/frontend/trace/ops/gather.py:118]: We should write a directed test instead of changing the example IMO

Comment (parthchadha) [reply tripy/tripy/frontend/trace/ops/gather.py:118]: I filed https://github.com/NVIDIA/TensorRT-Incubator/issues/52 to add a lot more exhaustive tests.

**Review by pranavm-nvidia — APPROVED** [review]

**Review by jhalakpatel — APPROVED** [review]

**Comment by jhalakpatel** [comment]
@parthchadha I thought we decided to use [`tripy`](https://github.com/NVIDIA/TensorRT-Incubator/labels/tripy) instead of `[tripy]`? I plan to add this guideline to `CONTRIBUTING.md`.

**Pull request merged**